File: app/src/main/java/com/example/simplerecyclerview/ItemArrayAdapter.java
package com.example.simplerecyclerview;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

public class ItemArrayAdapter extends RecyclerView.Adapter<ItemArrayAdapter.ViewHolder> {

    //All methods in this adapter are required for a bare minimum recyclerview adapter
    private int listItemLayout;
    private ArrayList<Item> itemList;

    // Constructor of the class
    public ItemArrayAdapter(int layoutId, ArrayList<Item> itemList) {
        listItemLayout = layoutId;
        this.itemList = itemList;
    }

    // get the size of the list
    @Override
    public int getItemCount() {
        return itemList == null ? 0 : itemList.size();
    }


    // specify the row layout file and click for each row
    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(listItemLayout, parent, false);
        ViewHolder myViewHolder = new ViewHolder(view);
        return myViewHolder;
    }

    // load data in each row element
    @Override
    public void onBindViewHolder(final ViewHolder holder, final int listPosition) {
        TextView name = holder.name;
        TextView title = holder.title;
        TextView id = holder.id;
        name.setText(itemList.get(listPosition).getName());
        title.setText(itemList.get(listPosition).getTitle());
        id.setText(itemList.get(listPosition).getId());
    }

    // Static inner class to initialize the views of rows
    static class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
        public TextView name,title,id;

        public ViewHolder(View itemView) {
            super(itemView);
            itemView.setOnClickListener(this);
            name = (TextView) itemView.findViewById(R.id.name);
            title = (TextView) itemView.findViewById(R.id.title);
            id = (TextView) itemView.findViewById(R.id.id);
        }

        @Override
        public void onClick(View view) {
            Log.d("onclick", "onClick " + getLayoutPosition() + " " + name.getText());
        }
    }
}

File: app/src/main/java/com/example/simplerecyclerview/ItemArrayAdapter.kt
package com.example.simplerecyclerview

import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

class ItemArrayAdapter     // Constructor of the class
    (//All methods in this adapter are required for a bare minimum recyclerview adapter
    private val listItemLayout: Int, private val itemList: ArrayList<Item>?
) : RecyclerView.Adapter<ItemArrayAdapter.ViewHolder>() {
    // get the size of the list
    override fun getItemCount(): Int {
        return itemList?.size ?: 0
    }

    // specify the row layout file and click for each row
    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): ViewHolder {
        val view =
            LayoutInflater.from(parent.context).inflate(listItemLayout, parent, false)
        return ViewHolder(view)
    }

    // load data in each row element
    override fun onBindViewHolder(holder: ViewHolder, listPosition: Int) {
        val name = holder.name
        val title = holder.title
        val id = holder.id
        name.text = itemList!![listPosition].name
        title.text = itemList[listPosition].title
        id.text = itemList[listPosition].id
    }

    // Static inner class to initialize the views of rows
    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView), View.OnClickListener {
        var name: TextView
        var title: TextView
        var id: TextView
        override fun onClick(view: View) {
            Log.d("onclick", "onClick " + layoutPosition + " " + name.text)
        }

        init {
            itemView.setOnClickListener(this)
            name = itemView.findViewById<View>(R.id.name) as TextView
            title = itemView.findViewById<View>(R.id.title) as TextView
            id = itemView.findViewById<View>(R.id.id) as TextView
        }
    }
}